From 25c036ca59978e9c07a0a60b9efcbd7e99cae7b3 Mon Sep 17 00:00:00 2001
From: Carl Schwan <carl@carlschwan.eu>
Date: Sun, 7 Jul 2024 23:35:50 +0200
Subject: [PATCH] Make KirigamiPrivate a static lib

So we don't need to install it and add a soversion. This also simplify a
bit the code.
---
 src/CMakeLists.txt | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d6be28437..c668c6544 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -24,7 +24,7 @@ endif()
 
 # Module: org.kde.kirigami.private
 
-add_library(KirigamiPrivate)
+add_library(KirigamiPrivate STATIC)
 ecm_add_qml_module(KirigamiPrivate
     URI "org.kde.kirigami.private"
     GENERATE_PLUGIN_SOURCE
@@ -37,18 +37,15 @@ target_sources(KirigamiPrivate PRIVATE
     actionhelper.h
 )
 
-include_static_dependencies(KirigamiPrivate EXPORT KirigamiTargets)
-
 target_link_libraries(KirigamiPrivate PRIVATE Qt6::Gui)
-
-ecm_finalize_qml_module(KirigamiPrivate DESTINATION ${KDE_INSTALL_QMLDIR})
+set_target_properties(KirigamiPrivate PROPERTIES POSITION_INDEPENDENT_CODE ON)
+set_target_properties(KirigamiPrivateplugin PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
 if (NOT BUILD_SHARED_LIBS)
     install(TARGETS KirigamiPrivateplugin EXPORT KirigamiTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
+    install(TARGETS KirigamiPrivate EXPORT KirigamiTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
 endif()
 
-install(TARGETS KirigamiPrivate EXPORT KirigamiTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
-
 # Module: org.kde.kirigami
 
 ecm_add_qml_module(Kirigami URI "org.kde.kirigami" VERSION 2.0
@@ -263,6 +260,8 @@ target_link_libraries(Kirigami
     Qt6::Quick
     PRIVATE
     Qt6::Concurrent
+    KirigamiPrivate
+    KirigamiPrivateplugin
     ${Kirigami_EXTRA_LIBS}
 )
 
@@ -280,7 +279,6 @@ if (NOT BUILD_SHARED_LIBS)
             KirigamiPrimitivesplugin
             KirigamiDialogsplugin
             KirigamiLayoutsplugin
-            KirigamiPrivateplugin
     )
     install(TARGETS Kirigamiplugin EXPORT KirigamiTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
 else()
@@ -292,7 +290,6 @@ else()
             KirigamiPrimitives
             KirigamiDialogs
             KirigamiLayouts
-            KirigamiPrivate
     )
 endif()
 
-- 
GitLab

